src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -18629,5 +18629,510 @@ public static void Main()
                 Diagnostic(ErrorCode.ERR_RefConstraintNotSatisfied, "int").WithArguments("R2<T>", "T", "int").WithLocation(10, 20)
                 );
         }
+
+        [Fact]
+        public void AccessCheckProtected03()
+        {
+            CSharpCompilation c = CreateCompilation(@"
+record X<T> { }
+
+record A { }
+
+record B
+{
+    record C : X<C.D.E>
+    {
+        protected record D : A
+        {
+            public record E { }
+        }
+    }
+}
+");
+
+            c.VerifyDiagnostics(
+                    // (8,12): error CS0060: Inconsistent accessibility: base type 'X<B.C.D.E>' is less accessible than class 'B.C'
+                    //     record C : X<C.D.E>
+                    Diagnostic(ErrorCode.ERR_BadVisBaseClass, "C").WithArguments("B.C", "X<B.C.D.E>").WithLocation(8, 12),
+                    // (8,12): error CS0050: Inconsistent accessibility: return type 'X<B.C.D.E>' is less accessible than method 'B.C.<Clone>$()'
+                    //     record C : X<C.D.E>
+                    Diagnostic(ErrorCode.ERR_BadVisReturnType, "C").WithArguments("B.C.<Clone>$()", "X<B.C.D.E>").WithLocation(8, 12),
+                    // (8,12): error CS0051: Inconsistent accessibility: parameter type 'X<B.C.D.E>' is less accessible than method 'B.C.Equals(X<B.C.D.E>?)'
+                    //     record C : X<C.D.E>
+                    Diagnostic(ErrorCode.ERR_BadVisParamType, "C").WithArguments("B.C.Equals(X<B.C.D.E>?)", "X<B.C.D.E>").WithLocation(8, 12)
+                    );
+        }
+
+        [Fact]
+        public void TestTargetType_Abstract()
+        {
+            var source = @"
+abstract record C
+{
+    void M()
+    {
+        C x0 = new();
+        var x1 = (C)new();
+    }
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,16): error CS0144: Cannot create an instance of the abstract type 'C'
+                //         C x0 = new();
+                Diagnostic(ErrorCode.ERR_NoNewAbstract, "new()").WithArguments("C").WithLocation(6, 16),
+                // (7,21): error CS0144: Cannot create an instance of the abstract type 'C'
+                //         var x1 = (C)new();
+                Diagnostic(ErrorCode.ERR_NoNewAbstract, "new()").WithArguments("C").WithLocation(7, 21)
+                );
+        }
+
+        [Fact]
+        public void CyclicBases4()
+        {
+            var text =
+@"
+record A<T> : B<A<T>> { }
+record B<T> : A<B<T>> {
+    A<T> F() { return null; }
+}
+";
+            var comp = CreateCompilation(text);
+            comp.GetDeclarationDiagnostics().Verify(
+                // (2,8): error CS0146: Circular base class dependency involving 'B<A<T>>' and 'A<T>'
```

Comment:
>class [](start = 54, length = 5)

Is this accurate?

Reply:
I don't have a better idea. I'm planning to leave this as-is.